runner/bsp/worker/src/mill/bsp/worker/BspWorkerImpl.scala

```diff
@@ -93,4 +96,21 @@ object BspWorkerImpl {
         )
     }
   }
+
+  private val executorCounter = new AtomicInteger
+  private def createJsonrpcExecutor(): ExecutorService =
+    Executors.newCachedThreadPool(
```

Comment:
What's this for? I thought the PR description said we were trying to avoid using thread pools

Reply:
`MillBuildServer` itself doesn't use a thread pool any more, but the jsonrpc library still needs one. Before this PR, both shared a common thread pool, now it's specific to jsonrpc handling.

I think the jsonrpc library uses it to accept incoming connections, to call methods on our build server instance (even though they don't block), etc.